### PR TITLE
Refactored filters API

### PR DESCRIPTION
**Pull request by marcuslinke**
This PR eleminates the use of the generic `Filters` object in the command APIs.

### CODE REVIEW

**Comment by KostyaSha** [comment]
You can force push to existed branch + previous PR was mergeable.

File: src/main/java/com/github/dockerjava/jaxrs/ListNetworksCmdExec.java [hunk 1]
```diff
@@ -26,9 +29,9 @@ public ListNetworksCmdExec(WebTarget baseResource, DockerClientConfig dockerClie
     protected List<Network> execute(ListNetworksCmd command) {
         WebTarget webTarget = getBaseResource().path("/networks");
 
-        if (command.getFilters() != null) {
+        if (command.getFilters() != null && !command.getFilters().isEmpty()) {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/ListNetworksCmdExec.java]: `Collections.notEmpty()` or something from guava that already do null check?

**Comment by KostyaSha** [comment]
You are merging faster than i'm reviewing.

**Comment by marcuslinke** [comment]
@KostyaSha Sorry, I thought you reviewed original PR already.

**Comment by KostyaSha** [comment]
But new commits periodically appears new :)

**Comment by KostyaSha** [comment]
Note, for any null/empty/string length try search for helpers in guava/commons/collections.
With java 1.8 `Objects.isNull()` `Obects.nonNull()` will be available that will end to nicer code
```
if (isNull(someObject)) {
```
or 
``` 
  public void someMethod(String arg) {
      Objects.requireNonNull("Fail", arg);
```

**Pull request merged**